testing/integration/index.ts

```diff
@@ -0,0 +1,120 @@
+            if (!outputResult.success) {
+                return Promise.resolve({ success: false, error: outputResult.error })
+            }
+            return Promise.resolve({ success: true });;
+        } catch (e) {
+            return Promise.resolve({ success: false, error: e });
+        }
+    }
+
+    public async teardown(): Promise<RunnerResult> {
+        try {
+            const destoryResult = await this.destroy();
+            if (!destoryResult.success) {
+                return Promise.resolve({ success: false, error: destoryResult.error })
+            }
+            const rmResult = await this.stackRemove();
+            if (!rmResult.success) {
+                return Promise.resolve({ success: false, error: rmResult.error })
+            }
+            return Promise.resolve({ success: true });
+        } catch (e) {
+            return Promise.resolve({ success: false, error: e });
+        }
+    }
+
+    public getStackOutput(key: string): string {
+        return this.outputs[key]
+    };
+
+    public getStackOutputKeys(): string[] {
+        return Object.keys(this.outputs);
+    }
+
+    public getStackName(): string {
+        return this.stackName;
+    }
+
+    private async stackInit(): Promise<CommandResult> {
+        return await execCommand("pulumi", ["stack", "init", this.stackName, "--cwd", this.pulumiProjDir]);
+    }
+
+    private async up(): Promise<CommandResult> {
+        return await execCommand("pulumi", ["up", "--non-interactive", "--cwd", this.pulumiProjDir, ...this.config]);
+    }
+
+    private async stackOutputs(): Promise<CommandResult> {
+        try {
+            const result: string[] = [];
+            const onStdOut = (output: string) => {
+                result.push(output);
+            };
+            const outputResult = await execCommand("pulumi", ["stack", "output", "--cwd", this.pulumiProjDir, "--json"], undefined, onStdOut);
+            if (!outputResult.success) throw new Error("Failed to get stack outputs: ${outputResult.error}");
+
+            const json = result.join();
+            const parsedOutput = JSON.parse(json);
+            for (const key of Object.keys(parsedOutput)) {
+                this.outputs[key] = parsedOutput[key];
+            }
+            return Promise.resolve({ success: true });
+        } catch (e) {
+            return Promise.resolve({ success: false, error: e });
+        }
+
+    }
+
+    private async destroy(): Promise<CommandResult> {
+        return await execCommand("pulumi", ["destroy", "--non-interactive", "--cwd", this.pulumiProjDir]);
+    }
+
+    private async stackRemove(): Promise<CommandResult> {
+        return await execCommand("pulumi", ["stack", "rm", this.stackName, "--cwd", this.pulumiProjDir, "--yes"]);
+    }
+
+    private createConfigArray(config: { [key: string]: string }): string[] {
+        const result: string[] = [];
+        for (const key of Object.keys(config)) {
+            result.push("-c");
```

Comment:
Why the "-c" between every config line?

Reply:
As far as I can tell, there's no way to pass all of these values under one flag. For each config value you specify you have to use `-c`: `-c foo=bar -c fizz=buzz ...`